pwndbg/commands/cymbol.py

```diff
@@ -0,0 +1,253 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""
+Add, load, show, edit, or delete symbols for custom structures.
+
+For the generation of the symbols g++/gcc is being used under the hood.
+
+In case of remote debugging a binary which is not native to your architecture it
+is advised to configure the 'gcc-config-path' config parameter to your own cross-platform
+gnu gcc compiled toolchain for your target architecture.
+
+You are advised to configure the 'cymbol-editor' config parameter to the path of your
+favorite text editor. Otherwise cymbol exapnds $EDITOR and $VISUAL environement varialbes
+to find the path to the default text editor.
+"""
+
+import argparse
+import functools
+import os
+import subprocess
+import sys
+import tempfile
+
+import gdb
+
+import pwndbg
+import pwndbg.commands
+import pwndbg.gdblib.arch
+import pwndbg.lib.gcc
+import pwndbg.lib.tempfile
+from pwndbg.color import message
+
+gcc_compiler_path = pwndbg.gdblib.config.add_param(
+    "gcc-compiler-path",
+    "",
+    "Path to your own gnu gcc/g++ toolchain for generating imported symbols.",
+)
+
+cymbol_editor = pwndbg.gdblib.config.add_param(
+    "cymbol-editor", "", "Path to your editor of your choice for editing custom structures."
+)
+
+# Remeber loaded symbols. This would be useful for 'remove-symbol-file'.
+loaded_symbols = {}
+
+
+def unload_loaded_symbol(custom_structure_name):
+    custom_structure_symbols_file = loaded_symbols.get(custom_structure_name)
+    if custom_structure_symbols_file is not None:
+        gdb.execute(f"remove-symbol-file {custom_structure_symbols_file}")
+        loaded_symbols.pop(custom_structure_name)
+
+
+def OnlyWhenStructureExists(func):
```

Comment:
Shall we rename this to `OnlyWhenStructFileExists`? When reading tests I initially thought that this may be about "whether we have symbols for a given struct loaded into GDB".